phonejailOS/Views/AppListView.swift

```diff
@@ -0,0 +1,200 @@
+import SwiftUI
+
+struct AppListView: View {
+    @ObservedObject var viewModel: AppBlockViewModel
+    @State private var searchText = ""
+    @State private var selectedCategory: AppCategory?
+    
+    var body: some View {
+        NavigationView {
+            VStack {
+                categoryFilter
+                appList
+            }
+            .navigationTitle("Apps")
+            .toolbar { refreshToolbar }
+            .alert("Error", isPresented: .constant(viewModel.error != nil)) {
+                Button("OK") { viewModel.clearError() }
+            } message: {
+                if let error = viewModel.error {
+                    Text(error.localizedDescription)
+                }
+            }
+        }
+    }
+    
+    private var categoryFilter: some View {
+        ScrollView(.horizontal, showsIndicators: false) {
+            HStack {
+                CategoryButton(title: "All", isSelected: selectedCategory == nil) {
+                    selectedCategory = nil
+                }
+                ForEach(AppCategory.allCases, id: \ .self) { category in
+                    CategoryButton(
+                        title: category.rawValue.capitalized,
+                        isSelected: selectedCategory == category
+                    ) {
+                        selectedCategory = category
+                    }
+                }
+            }
+            .padding(.horizontal)
+        }
+        .padding(.vertical, 8)
+    }
+    
+    private var appList: some View {
+        List {
+            ForEach(filteredApps) { app in
+                AppRow(app: app, viewModel: viewModel)
+            }
+        }
+        .listStyle(.plain)
+        .searchable(text: $searchText, prompt: "Search apps")
+    }
+    
+    private var refreshToolbar: some ToolbarContent {
+        ToolbarItem(placement: .navigationBarTrailing) {
+            Button(action: {
+                Task { await viewModel.loadBlockedApps() }
+            }) {
+                Image(systemName: "arrow.clockwise")
+            }
+        }
+    }
+    
+    private var filteredApps: [AppBlock] {
+        let allApps = viewModel.blockedApps
+        let categoryApps: [AppBlock]
+        if let category = selectedCategory {
+            categoryApps = allApps.filter { $0.appInfo.category == category }
+        } else {
+            categoryApps = allApps
+        }
+        if searchText.isEmpty {
+            return categoryApps
+        } else {
+            return categoryApps.filter { $0.appInfo.name.localizedCaseInsensitiveContains(searchText) }
+        }
```

Comment:
```suggestion
       categoryApps = selectedCategory.map { category in
    allApps.filter { $0.appInfo.category == category }
} ?? allApps

return searchText.isEmpty ? categoryApps : categoryApps.filter {
    $0.appInfo.name.localizedCaseInsensitiveContains(searchText)
}
```

this is using map to filter allApps by the condition (current category in this case) within categoryApps. otherwise the ternary operator (??) is saying to use all apps if categoryApps is nil

the return statement is checking if search text is empty and if it is then we return categoryApps otherwise (:) we filter the categoryApps by search text